server/project/middleware.ts

```diff
@@ -0,0 +1,102 @@
+import type {Request, Response, NextFunction} from 'express';
+import {Types} from 'mongoose';
+import ProjectCollection from '../project/collection';
+
+/**
+ * Checks if a project with projectId is req.params exists
+ */
+const isProjectExists = async (req: Request, res: Response, next: NextFunction) => {
+  const validFormat = Types.ObjectId.isValid(req.params.projectId);
+  const project = validFormat ? await ProjectCollection.findOne(req.params.projectId) : '';
+  if (!project) {
+    res.status(404).json({
+      error: `Project with project ID ${req.params.projectId} does not exist.`
+    });
+    return;
+  }
+
+  next();
+};
+
+/**
+ * Checks if the content of the project in req.body is valid, i.e. project name is non-empty and < 50 char, dates and users are valid values
+ */
+const isValidProjectFields = async (req: Request, res: Response, next: NextFunction) => {
```

Comment:
1) Moved the definition of invitedUsers and scheduledDates to after the returns of project if error, not sure if you were envisioning a larger change than that. 

2) These are helpful functions, thanks! I do think checking for index though may be good in this case because that way we can provide more clear error messages to the user.